multi-agent/adapters/outbound/mongo/blueprint_version_repository.py

```
@@ -0,0 +1,230 @@
+    equal to any dict value without materialising the actual bytes.
+    """
+    doc = BlueprintVersionDocument.from_mongo_doc(raw)
+    if "spec_dict_snapshot" not in raw:
+        # Projection excluded the field — substitute the sentinel so that
+        # downstream equality checks (`== {}` and `== actual_spec`) both
+        # evaluate to True via _SpecNotLoaded.__eq__.
+        doc.spec_dict_snapshot = _SPEC_NOT_LOADED
+    return doc
+
+
+def _ensure_utc(dt):
```

Ruff (0.15.17)

[warning] Missing return type annotation for private function `_ensure_utc`

(ANN202)